cmd/releasego/update-azure-linux.go

```diff
@@ -255,7 +255,7 @@ func updateAzureLinux(p subcmd.ParseFunc) error {
 }
 
 func generateUpdateBranchNameFromAssets(assets *buildassets.BuildAssets) string {
-	return fmt.Sprintf("refs/heads/update-go-%s", assets.GoVersion().Full())
+	return fmt.Sprintf("refs/heads/dev/go-%s", assets.GoVersion().Full())
```

Comment:
We could use the current auth's identity dynamically, but I think picking a static name is fine:

```suggestion
	return fmt.Sprintf("refs/heads/dev/bot-for-go/go-%s", assets.GoVersion().Full())
```